src/analyser.cpp

```diff
@@ -1117,12 +1240,785 @@ void Analyser::AnalyserImpl::analyseEquationAst(const AnalyserEquationAstPtr &as
+            issueDescriptions.push_back("The unit of " + expression(ast->mPimpl->mOwnedRightChild)
+                                        + " is not dimensionless. "
+                                        + expressionUnits(ast->mPimpl->mOwnedRightChild, rightUserUnitsMaps) + ".");
+        }
+    } else if ((ast->mPimpl->mType == libcellml::AnalyserEquationAst::Type::AND)
+               || (ast->mPimpl->mType == libcellml::AnalyserEquationAst::Type::OR)
+               || (ast->mPimpl->mType == libcellml::AnalyserEquationAst::Type::XOR)
+               || (ast->mPimpl->mType == libcellml::AnalyserEquationAst::Type::NOT)
+               || (ast->mPimpl->mType == libcellml::AnalyserEquationAst::Type::EXP)
+               || (ast->mPimpl->mType == libcellml::AnalyserEquationAst::Type::LN)
+               || (ast->mPimpl->mType == libcellml::AnalyserEquationAst::Type::LOG)) {
+        bool isDimensionlessUnitsMaps = Analyser::AnalyserImpl::isDimensionlessUnitsMaps(unitsMaps);
+        bool isDimensionlessRightUnitsMaps = Analyser::AnalyserImpl::isDimensionlessUnitsMaps(rightUnitsMaps);
+
+        if (!isDimensionlessUnitsMaps || !isDimensionlessRightUnitsMaps) {
+            std::string issueDescription = "The unit";
+
+            if (!isDimensionlessUnitsMaps && !isDimensionlessRightUnitsMaps) {
+                issueDescription += "s";
+            }
+
+            issueDescription += " of ";
+
+            if (!isDimensionlessUnitsMaps) {
+                issueDescription += expression(ast->mPimpl->mOwnedLeftChild, false);
+            }
+
+            if (!isDimensionlessUnitsMaps && !isDimensionlessRightUnitsMaps) {
+                issueDescription += " and ";
+            }
+
+            if (!isDimensionlessRightUnitsMaps) {
+                issueDescription += expression(ast->mPimpl->mOwnedRightChild, false);
+            }
+
+            issueDescription += " in " + expression(ast);
+
+            if (!isDimensionlessUnitsMaps && !isDimensionlessRightUnitsMaps) {
+                issueDescription += " are ";
+            } else {
+                issueDescription += " is ";
+            }
+
+            issueDescription += "not dimensionless. ";
+
+            if (!isDimensionlessUnitsMaps) {
+                issueDescription += expressionUnits(ast->mPimpl->mOwnedLeftChild, userUnitsMaps);
+            }
+
+            if (!isDimensionlessUnitsMaps && !isDimensionlessRightUnitsMaps) {
+                issueDescription += " while ";
+            }
+
+            if (!isDimensionlessRightUnitsMaps) {
+                issueDescription += expressionUnits(ast->mPimpl->mOwnedRightChild, rightUserUnitsMaps);
+            }
+
+            issueDescription += ".";
+
+            issueDescriptions.push_back(issueDescription);
+        }
+    } else if ((ast->mPimpl->mType == AnalyserEquationAst::Type::TIMES)
+               || (ast->mPimpl->mType == libcellml::AnalyserEquationAst::Type::DIVIDE)) {
```

Comment:
I don't think we need the full qualification here of the namespace.

Reply:
You mean for `AnalyserEquationAst::Type::TIMES` and `libcellml::AnalyserEquationAst::Type::DIVIDE`? If so, then we definitely do.

Reply:
Oops! Fixed in commit 1e5300d9.